source/globalCommands.py

```diff
@@ -1428,6 +1428,23 @@ def script_toggleMouseTracking(self,gesture):
 	script_toggleMouseTracking.__doc__=_("Toggles the reporting of information as the mouse moves")
 	script_toggleMouseTracking.category=SCRCAT_MOUSE
 
+	def script_toggleMouseTextResolution(self,gesture):
+		values = textInfos.MOUSE_TEXT_RESOLUTION_UNITS
+		labels = [textInfos.unitLabels[x] for x in values]
+		try:
+			index = values.index(config.conf["mouse"]["mouseTextUnit"])
+		except:
+			index=0

```

Comment:
I think it's a good idea to log something here.

Reply:
I think it would be quite unlikely for this to raise, however could you please specify the exception type or types?

The default in config spec is 'paragraph'. At the moment, if this fails I think it will default to `UNIT_CHARACTER`. Could you make it match the config spec?

Please include in the log message the name of the mouse text unit.

Reply:
Actually, I notice below that the index is incremented (and wrapped) so I think this would default back to paragraph. Also the resolution is announced, so no need to add it to the log message.

Reply:
[**@feerrenrut**](https://github.com/feerrenrut) commented on [2 May 2019, 17:03 CEST](https://github.com/nvaccess/nvda/pull/9057#discussion_r280461543 "2019-05-02T15:03:32Z - Replied by Github Reply Comments"):
> Please include in the log message the name of the mouse text unit.

Just to make sure, what unit do you want me to include? The new or the old one?

Reply:
Ah, It's now clear to me. Actually index 0 would set the setting to character, you were initially right.